src/vivarium_public_health/disease/state.py

```diff
@@ -75,13 +71,24 @@ def __init__(
 
         self.event_time_column = self.state_id + "_event_time"
         self.event_count_column = self.state_id + "_event_count"
+        self.prevalence_pipeline = f"{self.state_id}.prevalence"
+        self.birth_prevalence_pipeline = f"{self.state_id}.birth_prevalence"
         self.prevalence = 0.0
         self.birth_prevalence = 0.0
 
     def setup(self, builder: Builder) -> None:
-        super().setup(builder)
         self.prevalence_table = self.build_lookup_table(builder, "prevalence")
+        builder.value.register_attribute_producer(
```

Comment:
Why do we now need to register these as attributes?

Reply:
Previously LookupTables weren't Resources. Now that they are, they need to be called out as resources that are needed during initialization for the `DiseaseModel` component. That happens during `setup()`, but we don't have guarantees of the ordering of setup calls, so the `DiseaseModel` might get called first, before the `LookupTable` has actually been created. We have affordances for dealing with Pipelines that haven't been created yet, but not for LookupTables. I don't think in general, components should be interacting with LookupTables that have been created by other components.